modular_skyrat/modules/customization/modules/clothing/clothing.dm

```diff
@@ -19,6 +19,154 @@
 	var/greyscale_config_worn_taur_paw
 	var/greyscale_config_worn_taur_hoof
 
+	/// has_taur_sprite[icon_state]: Does this icon_state have ANY taur sprite?
+	var/static/list/has_taur_sprite = list()
+	/// has_taur_snake_sprite[icon_state]: Does this icon_state have a snake taur sprite?
+	var/static/list/has_taur_snake_sprite = list()
+	/// has_taur_paw_sprite[icon_state]: Does this icon_state have a pawed taur sprite?
+	var/static/list/has_taur_paw_sprite = list()
+	/// has_taur_horse_sprite[icon_state]: Does this icon_state have a hooved taur sprite?
+	var/static/list/has_taur_horse_sprite = list()
+
+	/// has_taur_sprite_suit[icon_state]: has_taur_sprite but for /obj/item/clothng/suit
+	var/static/list/has_taur_sprite_suit = list()
+	/// has_taur_snake_sprite_suit[icon_state]: has_taur_snake_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_snake_sprite_suit = list()
+	/// has_taur_paw_sprite_suit[icon_state]: has_taur_paw_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_paw_sprite_suit = list()
+	/// has_taur_horse_sprite_suit[icon_state]: has_taur_horse_sprite but for /obj/item/clothing/suit
+	var/static/list/has_taur_horse_sprite_suit = list()
```

Comment:
Why do these lists need to exist? if you want fast lookup, couldn't you just make the sprites lists assoc = TRUE?

Reply:
by this I mean:
The `has_taur_*_sprite` lists are just the icon states in the global lists, but assoc to true instead.
Why not just generate those global lists to be assoc to true, and access them if you want to check if a sprite is valid or not.

Assoc lookup ought (I hope) to be constant time (s not like these lists are huge), and it would save a lot of cruft and duplicate code that you're doing here.